### ISSUE TITLE
Empty arguments list passed to `generate/3` in Elixir 1.16

### ISSUE THREAD

**Issue by tomtaylor** [cabol/nebulex#218]
We're experiencing an issue which emerged when testing an upgrade to Elixir 1.16. The arguments passed to `generate/3` in our `Nebulex.Caching.KeyGenerator` module are now always an empty list. The actual arguments are never included. The first call to the function will fix the return values for all subsequent calls until the cache entry expires.

We're using a decorator in this form:

```elixir
  @decorate cacheable(
              cache: @cache,
              opts: [ttl: :timer.hours(24)]
            )
  def thing?(a, b)
    ...
  end
```

I got the current test suite up and running on Elixir 1.16 in #217, so I can see it's not passing, and there's a number of other issues which may or may not be related.

**Comment by cabol** [comment]
Thank you for reporting the issue. I will check it out and fix it ASAP!

**Comment by cabol** [comment]
I pushed a fix to support Elixir 1.16. I will push a new release soon. Please try it out and feel free to reopen the issue in case of a problem.

**Comment by stocks29** [comment]
Yikes, that's a scary bug. Luckily our project's test suite caught this. I can confirm that the fix in commit 0886eff511a420e1e0d600cc369205ecb23cd7ec resolved the issue for us.

When can we expect a new release to hit hex?
